### PR TITLE
feat: connect component manager to platform

### PR DESCRIPTION
**Pull request by MksmOrlov**
CI will be fixed after platform update

* [x] Update changelog if needed
* [x] Update SCn documentation if needed

### CODE REVIEW

File: scripts/build_sc_component_manager.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	esac
+	shift 1
+done
+
+if ((build_force == 1));
+then
+  stage "Clear latest build"
+  rm -rf "${BIN_PATH}"
+  rm -rf "${BUILD_PATH}"
+  find "${SC_COMPONENT_MANAGER_PATH}" -type d -name generated -exec rm -rf {} +
```
Comment (NikitaZotov) [inline scripts/build_sc_component_manager.sh:61]: Are you sure that it works?

File: scripts/build_sc_component_manager.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      outer_cmake_args+=("${1}")
+      ;;
+	  "-h"|"--help" )
+	    usage
+      exit 0
+			;;
+    * )
+      echo -e "$(basename "$0"): unknown flag $1"
+      echo "Try '$0 -h' for help"
+      exit 2
```
Comment (NikitaZotov) [inline scripts/build_sc_component_manager.sh:50]: Fix format

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,39 +1,57 @@
...
-add_library(sc-component-manager-lib SHARED ${SOURCES})
+	# for std::thread support
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
+endif(${UNIX})
 
-include_directories(${GLIB2_INCLUDE_DIRS} ${SC_COMPONENT_MANAGER_ROOT}/../sc-config-utils)
-target_link_libraries(sc-component-manager-lib sc-memory sc-agents-common sc-builder-lib sc-agents-common)
-add_dependencies(sc-component-manager-lib sc-code-generator)
+set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -DSC_DEBUG -DSC_PROFILE")
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DSC_DEBUG -DSC_PROFILE")
```
Comment (NikitaZotov) [inline CMakeLists.txt:47]:
```suggestion
```

File: scripts/build_sc_component_manager.sh [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+release_mode="-DCMAKE_BUILD_TYPE=Release"
+
+stage "Build sc-component-manager"
+
+cd "${ROOT_CMAKE_PATH}"
+cmake -B "${BUILD_PATH}" "${ROOT_CMAKE_PATH}" ${build_release:+${release_mode}} "${outer_cmake_args[@]}"
+cmake --build "${BUILD_PATH}" -j"$(nproc)"
+
+stage "sc-component-manager built successfully"
```
Comment (NikitaZotov) [inline scripts/build_sc_component_manager.sh:72]: I think that we should write `Sc-component-manager` or `The sc-component-manager`

File: scripts/run_sc_component_manager.sh [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+while getopts 'c:b:ih' OPTION; do
+  case "$OPTION" in
+    c)
+      CONFIG_PATH="$OPTARG"
+      CONFIG_ABSOLUTE_PATH=$(cd "$(dirname "$CONFIG_PATH")" && pwd -P)/$(basename "$CONFIG_PATH")
+      ;;
+    b)
+      KB_SRC_PATH="$OPTARG"
+      $BINARY_PATH/sc-builder -f --clear -c $CONFIG_ABSOLUTE_PATH -i $KB_SRC_PATH -o ../kb.bin
+      ;;
```
Comment (NikitaZotov) [inline scripts/run_sc_component_manager.sh:24]: This option doesn't work. And i think, that sc-component-manager has not this business

File: src/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+file(GLOB_RECURSE SOURCES "*.cpp")
+file(GLOB_RECURSE HEADERS "*.hpp")
+list(REMOVE_ITEM SOURCES
+    "${SC_COMPONENT_MANAGER_SRC}/main.cpp"
+)
+include_directories(${SC_COMPONENT_MANAGER_SRC} ${GLIB2_INCLUDE_DIRS} ${SC_MEMORY_SRC})
+
+add_library(sc-component-manager-lib SHARED ${SOURCES} ${HEADERS})
+
+target_link_libraries(sc-component-manager-lib sc-memory sc-config-utils sc-agents-common sc-builder-lib)
```
Comment (NikitaZotov) [inline src/CMakeLists.txt:12]: I propose to remove dependence from sc-agents-common in the future

**Review by NikitaZotov — CHANGES_REQUESTED** [review]

File: src/CMakeLists.txt (hunk 6 above)
Comment (MksmOrlov) [reply src/CMakeLists.txt:12]: ok, in the future

File: scripts/build_sc_component_manager.sh (hunk 1 above)
Comment (MksmOrlov) [reply scripts/build_sc_component_manager.sh:61]: Yes

File: src/CMakeLists.txt (hunk 6 above)
Comment (NikitaZotov) [reply src/CMakeLists.txt:12]: Create issue for this

File: scripts/build_sc_component_manager.sh [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+      exit 0
+			;;
+    * )
+      echo -e "$(basename "$0"): unknown flag $1"
+      echo "Try '$0 -h' for help"
+      exit 2
+			;;
+	esac
+	shift 1
+done
```
Comment (NikitaZotov) [inline scripts/build_sc_component_manager.sh:54]: Please, fix tabs

File: src/CMakeLists.txt (hunk 6 above)
Comment (NikitaZotov) [reply src/CMakeLists.txt:12]: I dont' see issue

File: scripts/build_sc_component_manager.sh (hunk 7 above)
Comment (NikitaZotov) [reply scripts/build_sc_component_manager.sh]: image

File: scripts/run_sc_component_manager.sh [hunk 8]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+set -e
```
Comment (NikitaZotov) [inline scripts/run_sc_component_manager.sh:3]:
```suggestion
set -eo pipefail
```

**Review by NikitaZotov — APPROVED** [review]

**Pull request merged**